### PR TITLE
Disable gradient reduction for Lora Adapters when SP is enabled

### PR DESCRIPTION
**Pull request by anmolgupt**
# What does this PR do ?

Disable gradient reduction for the Lora Adapter when SP is enabled. The first ColumnParallelLinear layer for the Lora adapter should not reduce the gradients when SP is enabled.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/modules/common/megatron/adapters/parallel_adapters.py [hunk 1]
```diff
@@ -187,6 +189,7 @@ def __init__(
                 bias=False,
                 gather_output=True,
                 init_method=self._get_init_fn(column_init_method),
+                disable_grad_reduce=self.disable_grad_reduce,
```
Comment (cuichenx) [inline nemo/collections/nlp/modules/common/megatron/adapters/parallel_adapters.py:192]: can this line be simply `disable_grad_reduce=self._sequence_parallel` ?

**Comment by cuichenx** [comment]
jenkins

**Comment by ericharper** [comment]
jenkins

**Comment by cuichenx** [comment]
I think we need to add an mcore version guard and/or change the megatron-lm commit in jenkinsfile

**Pull request closed without merging**